Fluid/Ast/IncludeStatement.cs

```diff
@@ -56,38 +60,56 @@ public IncludeStatement(Expression path, Expression with = null, IList<AssignSta
             {
                 var childScope = context.EnterChildScope();
                 string partialTemplate = await streamReader.ReadToEndAsync();
-                var parser = CreateParser(context);
-                if (parser.TryParse(partialTemplate, out var statements, out var errors))
+                var template = ParseTemplate(relativePath, fileProvider, context);
+                if (With != null)
                 {
-                    var template = CreateTemplate(context, statements);
-                    if (With != null)
-                    {
-                        var identifier = System.IO.Path.GetFileNameWithoutExtension(relativePath);
-                        var with = await With.EvaluateAsync(context);
-                        childScope.SetValue(identifier, with);
-                    }
+                    var identifier = System.IO.Path.GetFileNameWithoutExtension(relativePath);
+                    var with = await With.EvaluateAsync(context);
+                    childScope.SetValue(identifier, with);
+                }
 
-                    if (AssignStatements != null)
+                if (AssignStatements != null)
+                {
+                    foreach (var assignStatement in AssignStatements)
                     {
-                        foreach (var assignStatement in AssignStatements)
-                        {
-                            await assignStatement.WriteToAsync(writer, encoder, context);
-                        }
+                        await assignStatement.WriteToAsync(writer, encoder, context);
                     }
-
-                    await template.RenderAsync(writer, encoder, context);
-                }
-                else
-                {
-                    throw new Exception(String.Join(Environment.NewLine, errors));
                 }
 
+                await template.RenderAsync(writer, encoder, context);
+
                 childScope.ReleaseScope();
             }
 
             return Completion.Normal;
         }
 
+        private static IFluidTemplate ParseTemplate(string path, IFileProvider fileProvider, TemplateContext context)
+        {
+            return _memoryCache.GetOrCreate(path, entry =>
+            {
+                var fileInfo = fileProvider.GetFileInfo(path);
+                entry.SlidingExpiration = TimeSpan.FromHours(1);
```

Comment:
Static value.

Reply:
We can introduce an option to control this for both included templates and other liquid templates if it's needed

Reply:
We could but we need to enforce there is one for security reasons. If you can make it an option too then good.

Reply:
Ok